<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.15"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>wtdbg: &lt;a name=&quot;start&quot;&gt;&lt;/a&gt;Getting Started</title>
<link href="../../tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../jquery.js"></script>
<script type="text/javascript" src="../../dynsections.js"></script>
<link href="../../navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../resize.js"></script>
<script type="text/javascript" src="../../navtreedata.js"></script>
<script type="text/javascript" src="../../navtree.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
  $(document).ready(initResizable);
/* @license-end */</script>
<link href="../../search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../search/searchdata.js"></script>
<script type="text/javascript" src="../../search/search.js"></script>
<link href="../../doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">wtdbg
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.15 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "../../search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="../../menudata.js"></script>
<script type="text/javascript" src="../../menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('../../',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(document).ready(function(){initNavTree('d3/dcc/md__r_e_a_d_m_e.html','../../');});
/* @license-end */
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title"><a class="anchor" id="start"></a>Getting Started </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><div class="fragment"><div class="line">git clone https://github.com/ruanjue/wtdbg2</div><div class="line">cd wtdbg2 &amp;&amp; make</div><div class="line"># assemble long reads</div><div class="line">./wtdbg2 -x rs -g 4.6m -i reads.fa.gz -t16 -fo prefix</div><div class="line"># derive consensus</div><div class="line">./wtpoa-cns -t16 -i prefix.ctg.lay.gz -fo prefix.ctg.fa</div><div class="line"></div><div class="line"># polish consensus, not necessary if you want to polish the assemblies using other tools</div><div class="line">minimap2 -t 16 -x map-pb -a prefix.ctg.fa reads.fa.gz | samtools view -Sb - &gt;prefix.ctg.map.bam</div><div class="line">samtools sort prefix.ctg.map.bam prefix.ctg.map.srt</div><div class="line">samtools view prefix.ctg.map.srt.bam | ./wtpoa-cns -t 16 -d prefix.ctg.fa -i - -fo prefix.ctg.2nd.fa</div><div class="line"></div><div class="line"># polish contigs using short reads</div><div class="line">bwa mem -t 16 prefix.ctg.fa sr.1.fa sr.2.fa | samtools view -Sb - &gt;sr.bam</div><div class="line">samtools sort sr.bam sr.srt</div><div class="line">samtools view sr.srt.bam | ./wtpoa-cns -t 16 -x sam-sr -d prefix.ctg.fa -i - -fo prefix.ctg.3rd.fa</div></div><!-- fragment --><h2><a class="anchor" id="intro"></a>Introduction</h2>
<p>Wtdbg2 is a <em>de novo</em> sequence assembler for long noisy reads produced by PacBio or Oxford Nanopore Technologies (ONT). It assembles raw reads without error correction and then builds the consensus from intermediate assembly output. Wtdbg2 is able to assemble the human and even the 32Gb <a href="https://www.nature.com/articles/nature25458">Axolotl</a> genome at a speed tens of times faster than <a href="https://github.com/marbl/canu">CANU</a> and <a href="https://github.com/PacificBiosciences/FALCON">FALCON</a> while producing contigs of comparable base accuracy.</p>
<p>During assembly, wtdbg2 chops reads into 1024bp segments, merges similar segments into a vertex and connects vertices based on the segment adjacency on reads. The resulting graph is called fuzzy Bruijn graph (<a class="el" href="../../d3/d6c/struct_f_b_g.html">FBG</a>). It is akin to De Bruijn graph but permits mismatches/gaps and keeps read paths when collapsing k-mers. The use of <a class="el" href="../../d3/d6c/struct_f_b_g.html">FBG</a> distinguishes wtdbg2 from the majority of long-read assemblers.</p>
<h2><a class="anchor" id="install"></a>Installation</h2>
<p>Wtdbg2 only works on 64-bit Linux. To compile, please type <code>make</code> in the source code directory. You can then copy <code>wtdbg2</code> and <code>wtpoa-cns</code> to your <code>PATH</code>.</p>
<p>Wtdbg2 also comes with an approxmimate read mapper <code>kbm</code>, a faster but less accurate consesus tool <code>wtdbg-cns</code> and many auxiliary scripts in the <code>scripts</code> directory.</p>
<h2><a class="anchor" id="use"></a>Usage</h2>
<p>Wtdbg2 has two key components: an assembler <b>wtdbg2</b> and a consenser <b>wtpoa-cns</b>. Executable <b>wtdbg2</b> assembles raw reads and generates the contig layout and edge sequences in a file "*prefix*.ctg.lay.gz". Executable <b>wtpoa-cns</b> takes this file as input and produces the final consensus in FASTA. A typical workflow looks like this: </p><div class="fragment"><div class="line">./wtdbg2 -x rs -g 4.6m -t 16 -i reads.fa.gz -fo prefix</div><div class="line">./wtpoa-cns -t 16 -i prefix.ctg.lay.gz -fo prefix.ctg.fa</div></div><!-- fragment --><p> where <code>-g</code> is the estimated genome size and <code>-x</code> specifies the sequencing technology, which could take value "rs" for PacBio RSII, "sq" for PacBio Sequel, "ccs" for PacBio CCS reads and "ont" for Oxford Nanopore. This option sets multiple parameters and should be <b>applied before other parameters</b>. When you are unable to get a good assembly, you may need to tune other parameters as follows.</p>
<p>Wtdbg2 combines normal k-mers and homopolymer-compressed (HPC) k-mers to find read overlaps. Option <code>-k</code> specifies the length of normal k-mers, while <code>-p</code> specifies the length of HPC k-mers. By default, wtdbg2 samples a fourth of all k-mers by their hashcodes. For data of relatively low coverage, you may increase this sampling rate by reducing <code>-S</code>. This will greatly increase the peak memory as a cost, though. Option <code>-e</code>, which defaults to 3, specifies the minimum read coverage of an edge in the assembly graph. You may adjust this option according to the overall sequencing depth, too. Option <code>-A</code> also helps relatively low coverage data at the cost of performance. For PacBio data, <code>-L5000</code> often leads to better assemblies emperically, so is recommended. Please run <code>wtdbg2 --help</code> for a complete list of available options or consult <a class="el" href="../../db/d44/md__r_e_a_d_m_e-ori.html">README-ori.md</a> for more help.</p>
<p>The following table shows various command lines and their resource usage for the assembly step:</p>
<table class="markdownTable">
<tr class="markdownTableHead">
<th class="markdownTableHeadLeft">Dataset  </th><th class="markdownTableHeadRight">GSize  </th><th class="markdownTableHeadRight">Cov  </th><th class="markdownTableHeadLeft">Asm options  </th><th class="markdownTableHeadRight">CPU asm  </th><th class="markdownTableHeadRight">CPU cns  </th><th class="markdownTableHeadRight">Real tot  </th><th class="markdownTableHeadRight">RAM   </th></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyLeft"><a href="http://www.cbcb.umd.edu/software/PBcR/data/selfSampleData.tar.gz">E. coli</a>  </td><td class="markdownTableBodyRight">4.6Mb  </td><td class="markdownTableBodyRight">PB x20  </td><td class="markdownTableBodyLeft">-x rs -g4.6m -t16  </td><td class="markdownTableBodyRight">53s  </td><td class="markdownTableBodyRight">8m54s  </td><td class="markdownTableBodyRight">42s  </td><td class="markdownTableBodyRight">1.0G   </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyLeft"><a href="https://github.com/PacificBiosciences/DevNet/wiki/C.-elegans-data-set">C. elegans</a>  </td><td class="markdownTableBodyRight">100Mb  </td><td class="markdownTableBodyRight">PB x80  </td><td class="markdownTableBodyLeft">-x rs -g100m -t32  </td><td class="markdownTableBodyRight">1h07m  </td><td class="markdownTableBodyRight">5h06m  </td><td class="markdownTableBodyRight">13m42s  </td><td class="markdownTableBodyRight">11.6G   </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyLeft"><a href="https://www.ebi.ac.uk/ena/data/view/SRR5439404">D. melanogaster A4</a>  </td><td class="markdownTableBodyRight">144m  </td><td class="markdownTableBodyRight">PB x120  </td><td class="markdownTableBodyLeft">-x rs -g144m -t32  </td><td class="markdownTableBodyRight">2h06m  </td><td class="markdownTableBodyRight">5h11m  </td><td class="markdownTableBodyRight">26m17s  </td><td class="markdownTableBodyRight">19.4G   </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyLeft"><a href="https://www.ebi.ac.uk/ena/data/view/SRR6702603">D. melanogaster ISO1</a>  </td><td class="markdownTableBodyRight">144m  </td><td class="markdownTableBodyRight">ONT x32  </td><td class="markdownTableBodyLeft">-xont -g144m -t32  </td><td class="markdownTableBodyRight">5h12m  </td><td class="markdownTableBodyRight">4h30m  </td><td class="markdownTableBodyRight">25m59s  </td><td class="markdownTableBodyRight">17.3G   </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyLeft"><a href="https://downloads.pacbcloud.com/public/SequelData/ArabidopsisDemoData/">A. thaliana</a>  </td><td class="markdownTableBodyRight">125Mb  </td><td class="markdownTableBodyRight">PB x75  </td><td class="markdownTableBodyLeft">-x sq -g125m -t32  </td><td class="markdownTableBodyRight">11h26m  </td><td class="markdownTableBodyRight">4h57m  </td><td class="markdownTableBodyRight">49m35s  </td><td class="markdownTableBodyRight">25.7G   </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyLeft">Human NA12878  </td><td class="markdownTableBodyRight">3Gb  </td><td class="markdownTableBodyRight">ONT x36  </td><td class="markdownTableBodyLeft">-x ont -g3g -t31  </td><td class="markdownTableBodyRight">793h11m  </td><td class="markdownTableBodyRight">97h46m  </td><td class="markdownTableBodyRight">31h03m  </td><td class="markdownTableBodyRight">221.8G   </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyLeft"><a href="https://www.ebi.ac.uk/ena/data/view/PRJEB26791">Human NA19240</a>  </td><td class="markdownTableBodyRight">3Gb  </td><td class="markdownTableBodyRight">ONT x35  </td><td class="markdownTableBodyLeft">-x ont -g3g -t31  </td><td class="markdownTableBodyRight">935h31m  </td><td class="markdownTableBodyRight">89h17m  </td><td class="markdownTableBodyRight">35h20m  </td><td class="markdownTableBodyRight">215.0G   </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyLeft"><a href="https://www.ebi.ac.uk/ena/data/view/SRR7615963">Human HG00733</a>  </td><td class="markdownTableBodyRight">3Gb  </td><td class="markdownTableBodyRight">PB x93  </td><td class="markdownTableBodyLeft">-x sq -g3g -t47  </td><td class="markdownTableBodyRight">2114h26m  </td><td class="markdownTableBodyRight">152h24m  </td><td class="markdownTableBodyRight">52h22m  </td><td class="markdownTableBodyRight">338.1G   </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyLeft"><a href="https://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/PacBio_CCS_15kb/">Human NA24385</a>  </td><td class="markdownTableBodyRight">3Gb  </td><td class="markdownTableBodyRight">CCS x28  </td><td class="markdownTableBodyLeft">-x ccs -g3g -t31  </td><td class="markdownTableBodyRight">231h25m  </td><td class="markdownTableBodyRight">58h48m  </td><td class="markdownTableBodyRight">10h14m  </td><td class="markdownTableBodyRight">112.9G   </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyLeft"><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP044331">Human CHM1</a>  </td><td class="markdownTableBodyRight">3Gb  </td><td class="markdownTableBodyRight">PB x60  </td><td class="markdownTableBodyLeft">-x rs -g3g -t96  </td><td class="markdownTableBodyRight">105h33m  </td><td class="markdownTableBodyRight">139h24m  </td><td class="markdownTableBodyRight">5h17m  </td><td class="markdownTableBodyRight">225.1G   </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyLeft"><a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA378970">Axolotl</a>  </td><td class="markdownTableBodyRight">32Gb  </td><td class="markdownTableBodyRight">PB x32  </td><td class="markdownTableBodyLeft">-x rs -g32g -t96  </td><td class="markdownTableBodyRight">2806h40m  </td><td class="markdownTableBodyRight">1456h13m  </td><td class="markdownTableBodyRight">110h16m  </td><td class="markdownTableBodyRight">1788.1G   </td></tr>
</table>
<p>The timing was obtained on three local servers with different hardware configurations. There are also run-to-run fluctuations. Exact timing on your machines may differ. The assembled contigs can be found at the following FTP: </p><div class="fragment"><div class="line">ftp:<span class="comment">//ftp.dfci.harvard.edu/pub/hli/wtdbg/</span></div></div><!-- fragment --><h2>Limitations</h2>
<ul>
<li>For Nanopore data, wtdbg2 may produce an assembly smaller than the true genome.</li>
<li>When inputing multiple files of both fasta and fastq format, please put fastq first, then fasta. Otherwise, program cannot find '&gt;' in fastq, and append all fastq in one read.</li>
</ul>
<h2>Citing wtdbg2</h2>
<p>If you use wtdbg2, please cite:</p>
<blockquote class="doxtable">
<p>Ruan, J. and Li, H. (2019) Fast and accurate long-read assembly with wtdbg2. <em>bioRxiv</em>. doi:10.1101/530972 </p>
</blockquote>
<h2>Getting Help</h2>
<p>Please use the <a href="https://github.com/ruanjue/wtdbg2/issues">GitHub's Issues page</a> if you have questions. You may also directly contact Jue Ruan at <a href="#" onclick="location.href='mai'+'lto:'+'rua'+'nj'+'ue@'+'gm'+'ail'+'.c'+'om'; return false;">ruanj<span style="display: none;">.nosp@m.</span>ue@g<span style="display: none;">.nosp@m.</span>mail.<span style="display: none;">.nosp@m.</span>com</a>. </p>
</div></div><!-- PageDoc -->
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="../../doxygen.png" alt="doxygen"/></a> 1.8.15 </li>
  </ul>
</div>
</body>
</html>
