<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.15"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>wtdbg: NEWS</title>
<link href="../../tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../jquery.js"></script>
<script type="text/javascript" src="../../dynsections.js"></script>
<link href="../../navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../resize.js"></script>
<script type="text/javascript" src="../../navtreedata.js"></script>
<script type="text/javascript" src="../../navtree.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
  $(document).ready(initResizable);
/* @license-end */</script>
<link href="../../search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../search/searchdata.js"></script>
<script type="text/javascript" src="../../search/search.js"></script>
<link href="../../doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">wtdbg
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.15 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "../../search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="../../menudata.js"></script>
<script type="text/javascript" src="../../menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('../../',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(document).ready(function(){initNavTree('db/d44/md__r_e_a_d_m_e-ori.html','../../');});
/* @license-end */
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">NEWS </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><ul>
<li>|2018-09-21| Rename wtdbg-1.2.8 to wtdbg2</li>
<li>|2018-09-19| GFA supports <div class="fragment"><div class="line">wtdbg-dot2gfa.pl dbg.3.dot &gt;dbg.3.gfa</div></div><!-- fragment --></li>
<li>|2018-09-01| New consensus module: POACNS <br />
 <div class="fragment"><div class="line">wtpoa-cns -t 0 -i dbg.ctg.lay -fo dbg.ctg.poacns.fa</div></div><!-- fragment --> 1, wtpoa-cns implements POA to generate MSA from reads fragments within an edge. Used SSE instructions, but seems still can be further improved in speed <br />
 2, performs realignment on MSA. <br />
 3, recalibrates homopolymers, and make consensus sequence for an edge. Most reamining small errors come from homopolymer, more efforts are needed <br />
 4, joins edges' sequences into a contig. <br />
</li>
</ul>
<p>Welcome to test POACNS and feedback.</p>
<h1>WTDBG</h1>
<p>A fuzzy Bruijn graph (<a class="el" href="../../d3/d6c/struct_f_b_g.html">FBG</a>) approach to long noisy reads assembly</p>
<h1>Introduction</h1>
<p>A challenge in assembling long noisy reads from third generation sequencing (TGS) is reducing its requirement of computing resource, especially for large genomes. To address this issue, I developed a novel sequence alignment algorithm and a new assembly graph for efficiently assembling large genomes using TGS data.</p>
<ul>
<li>Alignment <br />
 <a class="el" href="../../d8/d6e/struct_k_b_m.html">KBM</a>: Kmer-BIN-Mapping.<br />
 <a class="el" href="../../d8/d6e/struct_k_b_m.html">KBM</a> groups k-mers from each non-overlapped sliding 256 bp fragments in long reads into bins. Bins of which most k-mers are high frequency, are filtered as highly repetitive ones. Then, <a class="el" href="../../d8/d6e/struct_k_b_m.html">KBM</a> searches synteny of matched bin pairs in sequences in a dynamic programming way. A matched bin pair in two sequences is defined as two bins different by original but share a set of k-mers. The result of alignments in <a class="el" href="../../d8/d6e/struct_k_b_m.html">KBM</a> have the same features of traditional sequence alignment, excepting the unit of <a class="el" href="../../d8/d6e/struct_k_b_m.html">KBM</a> alignments is 256 bp bin instead of single base.</li>
<li>Assembly <br />
 <a class="el" href="../../d3/d6c/struct_f_b_g.html">FBG</a>: Fuzzy Bruijn <a class="el" href="../../d4/dd4/struct_graph.html">Graph</a>. <br />
 <a class="el" href="../../d3/d6c/struct_f_b_g.html">FBG</a> is composed of vertices in length of 1024 bp from reads, and edges connecting vertices in their order on read paths. Comparing with <a class="el" href="../../d2/d39/struct_d_b_g.html">DBG</a>, the size of vertices in <a class="el" href="../../d3/d6c/struct_f_b_g.html">FBG</a> are much bigger, thus won¡¯t be sensitive to small repeat. To tolerate high sequencing errors, <a class="el" href="../../d3/d6c/struct_f_b_g.html">FBG</a>'s vertices are found using gapped sequence alignments from <a class="el" href="../../d8/d6e/struct_k_b_m.html">KBM</a> or other aligners, comparing with searching identical k-mers in <a class="el" href="../../d2/d39/struct_d_b_g.html">DBG</a>.</li>
<li>Why choose wtdbg in genome assembly <br />
 There are many assemblers for long noisy reads assembly, e.g. FALCON, CANU, miniasm, and SMARTdenovo (progenitor of wtdbg). If you have a genome of 10G bp or bigger in size, wtdbg is your first or even the only option. For small but complicated genomes (&lt; 3 G), wtdbg was often reported to yield better assembly by my friends. Besides, <a class="el" href="../../d8/d6e/struct_k_b_m.html">KBM</a> is easy to use when you are setting up a web-server for long reads mapping (see Example 2).</li>
<li>Limitation <br />
 Max read length is 0x0003FFFFU (256 Kb), longer reads will be split. <br />
 Max number of reads is 0x03FFFFFFU (64 M). If your data volume exceeds, please filter relative shorter reads. <br />
 In <a class="el" href="../../d8/d6e/struct_k_b_m.html">KBM</a>, max read length is 0xFFFFFFFFU (4 Gb), max number of reads is 0x0FFFFFFFU (256 M). <br />
 Max number of threads is 4096. <br />
 Cannot parallelly run in multiple nodes. But you can implement it simplely using <code>kbm</code> and <code>wtdbg --load-alignments</code> <br />
 Developed and tested in Linux-GCC only. <br />
 Only accepts fasta/fastq format for input, '.gz' suffixed files will be piped by <code>gzip -dc</code>.</li>
</ul>
<p># Installation </p><div class="fragment"><div class="line">git clone https://github.com/ruanjue/wtdbg2.git</div><div class="line">cd wtdbg2</div><div class="line">make</div></div><!-- fragment --><h1>Long reads mapping</h1>
<p>Supposes you have <code>hg19.fa</code> as reference sequences, and <code>reads.fq.gz</code> as query sequences. ## Example 1 </p><div class="fragment"><div class="line">kbm -t 64 -d hg19.fa -i reads.fa.gz -o reads.kbmap</div></div><!-- fragment --> <h3>output format</h3>
<ul>
<li>COL1 <code>qry_name</code></li>
<li>COL2 <code>qry_strand</code></li>
<li>COL3 <code>qry_length</code></li>
<li>COL4 <code>qry_beg</code></li>
<li>COL5 <code>qry_end</code></li>
<li>COL6 <code>ref_name</code></li>
<li>COL7 <code>ref_strand</code> (always equals <code>+</code>)</li>
<li>COL8 <code>ref_length</code></li>
<li>COL9 <code>ref_beg</code></li>
<li>COL10 <code>ref_end</code></li>
<li>COL11 <code>match_len</code> (length of matched k-mers)</li>
<li>COL12 <code>align_len</code> (length of aligned)</li>
<li>COL13 <code>#kcnt</code> (number of matched k-mers)</li>
<li>COL14 <code>#gap</code> (number of gapped BINs)</li>
<li>COL15 <code>cigar</code> (256 x SAM's cigar)</li>
</ul>
<h2>Example 2</h2>
<p>Suitable for online tools, or frequently used references ### Build KBM-INDEX </p><div class="fragment"><div class="line">kbm -t 64 -d hg19.fa -i /dev/null -W hg19.kbmidx</div></div><!-- fragment --><p> ### Set up <a class="el" href="../../d8/d6e/struct_k_b_m.html">KBM</a> server </p><div class="fragment"><div class="line">kbm -R hg19.kbmidx start</div></div><!-- fragment --><p> Now, hg19.kbmidx is cached in memory for further call.</p>
<p>### Mapping with KBM-INDEX </p><div class="fragment"><div class="line">kbm -R hg19.kbmidx -t 64 -i reads.fa -o reads.kbmap</div></div><!-- fragment --><p> Please note that, hg19.kbmidx can be multilple called by any processes in the same computer. <br />
</p>
<p>### Shutdown <a class="el" href="../../d8/d6e/struct_k_b_m.html">KBM</a> server </p><div class="fragment"><div class="line">kbm -R hg19.kbmidx stop</div></div><!-- fragment --><h1>Long reads assembling</h1>
<p>## Quick Start </p><div class="fragment"><div class="line">echo &quot;manual&quot;</div><div class="line">run_wtdbg_assembly.sh -h</div><div class="line">echo &quot;generating shell script&quot;</div><div class="line">run_wtdbg_assembly.sh -t 0 -i reads.fa.gz -o dbg -T &gt;run.sh</div></div><!-- fragment --><p> <code>run.sh</code> is ready for invoked</p>
<p>## Play with wtdbg </p><div class="fragment"><div class="line">wtdbg2 -h</div><div class="line">wtdbg2 --help</div></div><!-- fragment --><p> ### options </p><div class="fragment"><div class="line">-t &lt;int&gt;    Number of threads, 0: all cores, [0]</div><div class="line">-i &lt;string&gt; Long reads sequences file, + *</div><div class="line">-I &lt;string&gt; Error-free sequences file, +</div><div class="line">-o &lt;string&gt; Prefix of output files, *</div><div class="line">-f          Force overwrite</div><div class="line">-k &lt;int&gt;    Kmer fsize, 0 &lt;= k &lt;= 25, [0]</div><div class="line">-p &lt;int&gt;    Kmer psize, 0 &lt;= p &lt;= 25, [21]</div><div class="line">            k + p &lt;= 25, seed is &lt;k-mer&gt;+&lt;p-homopolymer-compressed&gt;</div><div class="line">-K &lt;float&gt;  Filter high frequency kmers, maybe repetitive, [1000]</div><div class="line">            if K &gt;= 1, take the integer value as cutoff</div><div class="line">            else, mask the top fraction part high frequency kmers</div><div class="line">-E &lt;int&gt;    Min kmer frequency, [2]</div><div class="line">-F          Filter low frequency kmers by a 4G-bytes array (max_occ=3 2-bits). Here, -E must greater than 1</div><div class="line">-S &lt;int&gt;    Subsampling kmers, 1/(&lt;-S&gt;) kmers are indexed, [4]</div><div class="line">            -S is very useful in saving memeory and speeding up</div><div class="line">            please note that subsampling kmers will have less matched length</div><div class="line">-X &lt;int&gt;    Max number of bin(256bp) in one gap, [4]</div><div class="line">-Y &lt;int&gt;    Max number of bin(256bp) in one deviation, [4]</div><div class="line">-x &lt;int&gt;    penalty for BIN gap, [-7]</div><div class="line">-y &lt;int&gt;    penalty for BIN deviation, [-21]</div><div class="line">-l &lt;float&gt;  Min length of alignment, [2048]</div><div class="line">-m &lt;float&gt;  Min matched, [200]</div><div class="line">-s &lt;float&gt;  Max length variation of two aligned fragments, [0.2]</div><div class="line">-q          Quiet</div><div class="line">-v          Verbose, +</div><div class="line">--help      Show more options</div></div><!-- fragment --><h3>For higher error rate long sequences</h3>
<p>Decrease <code>-p</code>. Try <code>-p 19</code> or <code>-p 17</code> <br />
 Decrease <code>-S</code>. Try <code>-S 2</code> or <code>-S 1</code> <br />
 Both will increase computing time.</p>
<h3>For very high coverage</h3>
<p>Increase <code>--edge-min</code>. Try <code>--edge-min 4</code>, or higher.</p>
<h3>For low coverage</h3>
<p>Decrease <code>--edge-min</code>. Try <code>--edge-min 2 --rescue-low-cov-edges</code>.</p>
<h3>Filter reads</h3>
<p><code>--tidy-reads 5000</code>. Will filtered shorter sequences. If names in format of <code>\/\d+_\d+$</code>, will selected the longest subread.</p>
<h3>output</h3>
<p>Suppose the prefix is <code>dbg</code></p><ul>
<li>dbg.1.dot <br />
 DOT file for initialized graph</li>
<li>dbg.1.nodes <br />
 nodes and their positions in reads</li>
<li>dbg.1.reads <br />
 reads and their nodes</li>
<li>dbg.2.dot <br />
 DOT file after transitive reduction</li>
<li>dbg.3.dot <br />
 DOT file after merging bubble and remove tips</li>
<li>dbg.alignments <br />
 KBMAP file, all vs all alignments</li>
<li>dbg.binkmer <br />
 Distribution of number of k-mers in a BIN</li>
<li>dbg.closed_bins <br />
 Filtered BINs</li>
<li>dbg.clps Reads clip information. <br />
 COL1 read_name <br />
 COL2 read_length <br />
 COL3 keep_offset <br />
 COL4 keep_length</li>
<li>dbg.ctg.dot <br />
 DOT file for contigs</li>
<li>dbg.ctg.lay <br />
 Contigs layout file. Will be read by <code>wtdbg-cns</code>. This file is the main result of <code>wtdbg</code> <br />
 <b>Format</b>:<br />
 <div class="fragment"><div class="line">&gt;ctg(\d+) nodes=(\d+) len=(\d+)</div><div class="line">E &lt;OFFSET&gt; &lt;NODE1&gt; &lt;STRAND1&gt; &lt;NODE2&gt; &lt;STRAND2&gt;</div><div class="line">S &lt;READ_NAME&gt; &lt;STRAND&gt; &lt;REG_OFFSET&gt; &lt;REG_LENGTH&gt; &lt;REG_SEQ&gt;</div><div class="line">S &lt;READ_NAME&gt; &lt;STRAND&gt; &lt;REG_OFFSET&gt; &lt;REG_LENGTH&gt; &lt;REG_SEQ&gt;</div><div class="line">S &lt;READ_NAME&gt; &lt;STRAND&gt; &lt;REG_OFFSET&gt; &lt;REG_LENGTH&gt; &lt;REG_SEQ&gt;</div><div class="line">...</div><div class="line">E ...</div><div class="line">...</div></div><!-- fragment --> One contig contains many edges (starting with 'E'), each edge contains many regions inside reads. <br />
 Please note that one read often contains many REGs.</li>
<li>dbg.events <br />
 Log file of graph simplification</li>
<li>dbg.frg.dot <br />
 DOT file for unitigs</li>
<li>dbg.frg.nodes <br />
 unitigs and their nodes</li>
<li>dbg.kmerdep <br />
 Distribution of k-mer depth</li>
<li>STDERR stream <br />
 wtdbg print runtime information on progrom's STDERR stream. <code>--quiet</code> to disiable it</li>
</ul>
<p>## Consensus </p><div class="fragment"><div class="line">wtdbg-cns -t 64 -i dbg.ctg.lay -o dbg.ctg.lay.fa</div></div><!-- fragment --><p> The output file <code>dbg.ctg.lay.fa</code> is ready for further polished by <code>PILON</code> or <code>QUIVER</code>.</p>
<div class="fragment"><div class="line">wtpoa-cns -t 64 -i dbg.ctg.lay -o dbg.ctg.lay.fa</div></div><!-- fragment --><p> wtpoa-cns is slower than wtdbg-cns, but offer more accurate consensus sequences. I will update it in following development.</p>
<h1>Performance</h1>
<h2>Human (3G) CHM1 PacBio P5C3 dataset, 65.5 core.hours</h2>
<ul>
<li>Data Source <a href="http://datasets.pacb.com/2014/Human54x/fasta.html">http://datasets.pacb.com/2014/Human54x/fasta.html</a></li>
<li>Command <div class="fragment"><div class="line">wtdbg2 -t 96 -i pb.fa -fo dbg --tidy-reads 5000 --edge-min 2 --rescue-low-cov-edges</div></div><!-- fragment --></li>
<li>Contigs <code>TOT 2978536704, CNT 8752, AVG 340327, MAX 11662848, N50 1925120, L50 453, N90 400128, L90 1727, Min 5120</code></li>
<li>Runtime <code>real 6131.803 sec, user 201836.200 sec, sys 33956.790 sec, maxrss 117281672.0 kB, maxvsize 202422172.0 kB</code></li>
</ul>
<h2>Human (3G) CHM1 PacBio P6C4 dataset, 211.3 core.hours</h2>
<ul>
<li>Data Source <a href="http://www.ebi.ac.uk/ena/data/view/PRJNA246220">http://www.ebi.ac.uk/ena/data/view/PRJNA246220</a></li>
<li>Command <div class="fragment"><div class="line">wtdbg2 -t 96 -i wt.fa -fo dbg --tidy-reads 5000 --edge-min 4 --rescue-low-cov-edges</div></div><!-- fragment --></li>
<li>Contigs <code>TOT 2964872448, CNT 1909, AVG 1553103, MAX 105310208, N50 23586816, L50 34, N90 3326976, L90 158, Min 5120</code></li>
<li>Runtime <code>real 16806.534 sec, user 681278.770 sec, sys 79371.630 sec, maxrss 264956752.0 kB, maxvsize 443356532.0 kB</code></li>
</ul>
<h2>Axolotl (32G) PacBio dataset, 32 X, 3053 core.hours</h2>
<ul>
<li>Command <div class="fragment"><div class="line">wtdbg2 -t 96 -i ../rawdata/pacbio.fa.gz -p 21 -S 2 --aln-noskip --rescue-low-cov-edges --tidy-reads 5000 -fo axolotl</div></div><!-- fragment --></li>
<li>Contigs <code>TOT 27375160576, CNT 115355, AVG 237313, MAX 7812608, N50 606976, L50 12527, N90 144896, L90 47295, Min 5120</code></li>
<li>Runtime <code>real 190237.591 sec, user 10994200.800 sec, sys 488715.030 sec, maxrss 1671005352.0 kB, maxvsize 2365400208.0 kB</code></li>
</ul>
<h2>Human (3G) NA12878 ONT dataset, 197.5 core.hours</h2>
<ul>
<li>Data Source <a href="https://github.com/nanopore-wgs-consortium/NA12878">https://github.com/nanopore-wgs-consortium/NA12878</a></li>
<li>Command <div class="fragment"><div class="line">wtdbg2 -t 64 -i NA12878-ONT.fa.gz -fo dbg -S 2 --edge-min 2 --rescue-low-cov-edges</div></div><!-- fragment --></li>
<li>Contigs <code>TOT 2827644928, CNT 19473, AVG 145209, MAX 31366400, N50 4540672, L50 162, N90 172800, L90 1111, Min 5120</code></li>
<li>Runtime <code>real 14992.925 sec, user 649202.270 sec, sys 61638.300 sec, maxrss 256840096.0 kB, maxvsize 356668088.0 kB</code></li>
</ul>
<h1>Citation</h1>
<p>To be published. <br />
 URL <b><a href="https://github.com/ruanjue/wtdbg2/">https://github.com/ruanjue/wtdbg2/</a></b> <br />
</p>
<h1>Contact</h1>
<p>Jue Ruan <a href="#" onclick="location.href='mai'+'lto:'+'rua'+'nj'+'ue@'+'gm'+'ail'+'.c'+'om'; return false;">ruanj<span style="display: none;">.nosp@m.</span>ue@g<span style="display: none;">.nosp@m.</span>mail.<span style="display: none;">.nosp@m.</span>com</a> <br />
 Jue Ruan <a href="#" onclick="location.href='mai'+'lto:'+'rua'+'nj'+'ue@'+'ca'+'as.'+'cn'; return false;">ruanj<span style="display: none;">.nosp@m.</span>ue@c<span style="display: none;">.nosp@m.</span>aas.c<span style="display: none;">.nosp@m.</span>n</a> </p>
</div></div><!-- PageDoc -->
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="../../doxygen.png" alt="doxygen"/></a> 1.8.15 </li>
  </ul>
</div>
</body>
</html>
